tests/test_signals.py

```diff
@@ -76,11 +77,28 @@ def verify_state(status, eager):
         assert status.state == UserTaskStatus.PENDING
 
 
+SIGNAL_DATA = {}
+
+
+def receiver(sender, **kwargs):  # pylint: disable=unused-argument
+    """
+    Signal handler for testing the user_task_stopped signal.
+    """
+    LOGGER.debug('Received signal')
+    SIGNAL_DATA['received_status'] = kwargs.get('status')
```

Comment:
If the signal gets called more than once, this receiver and the tests below won't detect it. Perhaps add a counter *or* some other way of ensuring that the signal is called only once?